server/app/mutations/grid_services/delete.rb

```diff
@@ -15,14 +15,29 @@ def validate
 
     def execute
       nodes = self.grid_service.grid_service_instances.map{ |i| i.host_node }
+      notify_lb_remove if self.grid_service.linked_to_load_balancer?
       self.grid_service.destroy
       nodes.each do |node|
         notify_node(node) if node
       end
     end
 
+    # @param [HostNode] node
     def notify_node(node)
       RpcClient.new(node.node_id).notify('/service_pods/notify_update', 'terminate')
     end
+
+    def notify_lb_remove
```

Comment:
Not optimal but I don't want to bloat this PR more than necessary.. old way of removing lb config just does not work with this new logic so some changes are necessary.